admin/database/database.go

```diff
@@ -276,6 +276,10 @@ type DB interface {
 	UpdateBillingIssueOverdueAsProcessed(ctx context.Context, id string) error
 	DeleteBillingIssue(ctx context.Context, id string) error
 	DeleteBillingIssueByTypeForOrg(ctx context.Context, orgID string, errorType BillingIssueType) error
+
+	FindProjectVariablesByEnvironment(ctx context.Context, projectID string, environment *string) ([]*ProjectVariable, error)
+	UpsertProjectVariable(ctx context.Context, projectID, environment string, vars map[string]string, userID string) ([]*ProjectVariable, error)
```

Comment:
As a reflection on this PR, I would consider breaking this into separate `InsertProjectVariable` and `UpdateProjectVariable` functions and making multiple calls from the API resolver.

The external API interface may pretend they are one set, but with multiple rows in the DB it makes the queries simpler (and performance/optimization is not really a concern on such small values anyway).

Reply:
I updated the `UpsertProjectVariable` implementation to avoid usage of `NamedQuery`. Do you think we still need separate APIs?
The upsert query itself is fairly simple and similar upserts are in-use at other places as well. If we are not worried about performance we can consider running one query per variable which will make it even simpler.
The reason I want to favour `upsert` over `insert`/`update` is that we will need to add logic for what database already provides.

Reply:
Okay it's fine to do upserts if you think it's simpler. It was mostly just a reflection on keeping things simple